### PR TITLE
Add wait_for_javascript to flaky system tests

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

I've added a helper called `wait_for_javascript` that waits for JS to be present in the page before starting the verification. In an ideal world it shouldn't be needed, but unfortunately Capybara and the wait it interacts with the page can lead to situations where the JS has not finished processing before it moves on the next step.

I've added this call using the list of system tests that tend to fail often lately, two perfect examples are the list of failures in these two builds:

- https://travis-ci.com/thepracticaldev/dev.to/builds/143557061#L1443
- https://travis-ci.com/thepracticaldev/dev.to/builds/143490153#L1009 

I got the idea from https://medium.com/doctolib/hunting-flaky-tests-2-waiting-for-ajax-bd76d79d9ee9

### CODE REVIEW

File: spec/support/initializers/capybara.rb [hunk 1]
```diff
@@ -28,3 +28,22 @@
...
 end
+
+# adapted from <https://medium.com/doctolib-engineering/hunting-flaky-tests-2-waiting-for-ajax-bd76d79d9ee9>
+def wait_for_javascript
+  max_time = Capybara::Helpers.monotonic_time + Capybara.default_max_wait_time
+  finished = false
+
+  while Capybara::Helpers.monotonic_time < max_time
+    begin
+      finished = page.evaluate_script("initializeBaseApp")
```
Comment (rhymes) [inline spec/support/initializers/capybara.rb]: I chose `initializeBaseApp` because it's the entry point of the client side JS

Comment (mstruve) [reply spec/support/initializers/capybara.rb]: Should we check if the javascript is done running first before we execute the initialize script ourselves? In the past I have used something like this to check for active running javascript. Not sure if something like that is an option here.
```ruby
  def wait_for_ajax
    wait_for { page.evaluate_script("$.active").zero? }
  end
```

Comment (rhymes) [reply spec/support/initializers/capybara.rb]: This doesn't run the initializer, it just checks if `initializeBaseApp` is defined, in previous apps I used to check for things like jQuery like in the blog post mentioned in the description but we don't use it. 

See an example:

```ruby
[1] pry(#<RSpec::ExampleGroups::EditingWithAnEditor>)> page.evaluate_script("initializeBaseApp")
Selenium::WebDriver::Error::JavascriptError: javascript error: initializeBaseApp is not defined
  (Session info: headless chrome=79.0.3945.117)
[2] pry(#<RSpec::ExampleGroups::EditingWithAnEditor>)> page.evaluate_script("window.navigator")
=> {"appCodeName"=>"Mozilla", ... }
```

I'm going to simplify the code removing the error handler, by using `typeof`:

```ruby
[16] pry(#<RSpec::ExampleGroups::EditingWithAnEditor>)> page.evaluate_script("typeof initializeBaseApp")
=> "undefined"
[17] pry(#<RSpec::ExampleGroups::EditingWithAnEditor>)> page.evaluate_script("typeof window.navigator")
=> "object"
```

**Comment by rhymes** [comment]
I'm comforted by the fact that the two builds of this PR haven't failed. Maybe we're onto something :D

File: spec/support/initializers/capybara.rb (hunk 1 above)
Comment (mstruve) [reply spec/support/initializers/capybara.rb]: Ah, I get it now! Thank you for rewriting it to be more clear for js noobs like myself 😊

**Pull request merged**